### PR TITLE
UX/UI for Stripe Elements

### PR DESCRIPTION
**Pull request by rethinkDeliriumRaj**
Non working site
For Testing:
- Create Account using 4242 4242 4242 4242 as the CC, should show success.
- Sign in using harsh@test.com, PW testtest

### CODE REVIEW

**Comment by rethinkDeliriumRaj** [comment]
@rethinkdeliriumjack please put eyes on this code change and let me know if anything doesn't make sense or looks wrong. Ask questions and we can hopefully make it a learning exp

File: components/CreateAccount/CreateAccountStripe.jsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
+import { Button } from "@mantine/core";
+import { useState, useContext } from "react";
+import { FormContext } from "../Context/FormContext";
+
+function CreateAccountStripe() {
+  const smallScreen = window.matchMedia("(max-width: 600px)").matches;
```
Comment (rethinkdeliriumjack) [inline components/CreateAccount/CreateAccountStripe.jsx:6]: How did you determine if the screen width is less than or equal to 600px is the right threshold? Industry standard?

File: components/CreateAccount/Questionnaire.jsx [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import { useContext, forwardRef, useState } from "react";
 import { FormContext } from "../Context/FormContext";
-import { Button, Group, Select, Text, TextInput } from "@mantine/core";
+import { Button, Group, PasswordInput, Select, Text, TextInput } from "@mantine/core";
 import { useForm } from "@mantine/form";
 
-function validateNewEmail(email) {
+async function createNewAccount(email) {
```
Comment (rethinkdeliriumjack) [inline components/CreateAccount/Questionnaire.jsx:6]: Why did you include async here?

File: components/CreateAccount/Questionnaire.jsx [hunk 3]
```diff
@@ -63,26 +63,28 @@ function Questionnaire() {
...
   const [loading, setLoading] = useState(false);
 
+  const handleSubmit = (values) => {
+    setLoading(true);
+    createNewAccount(values.caregiver.email).then((valid) => {
+      if (!valid) {
+        form.setFieldError("caregiver.email", "Email already in use");
+      }
+    });
+    setLoading(false);
```
Comment (rethinkdeliriumjack) [inline components/CreateAccount/Questionnaire.jsx:73]: Is it possible setLoading(false) is called immediately after the asynch function createNewAccount which might cause setLoading to be set to false before createNewAccount actually finishes?

File: components/CreateAccount/CreateAccountStripe.jsx (hunk 1 above)
Comment (rethinkDeliriumRaj) [reply components/CreateAccount/CreateAccountStripe.jsx:6]: Honestly just see it in all examples, and tested it around by playing w/ Chrome tools.

File: components/CreateAccount/Questionnaire.jsx (hunk 2 above)
Comment (rethinkDeliriumRaj) [reply components/CreateAccount/Questionnaire.jsx:6]: This in itself is not an async function, but in the future the goal is to access the backend here. So I wanted to prep it for treating it as async as opposed to instantaneous (EG - checking hhpatel is quick, but in the future might not be, so might as well treat it as slow now)

File: components/CreateAccount/Questionnaire.jsx [hunk 4]
```diff
@@ -63,26 +63,28 @@ function Questionnaire() {
   const smallScreen = window.matchMedia("(max-width: 600px)").matches;
   const [loading, setLoading] = useState(false);
 
+  const handleSubmit = (values) => {
+    setLoading(true);
+    createNewAccount(values.caregiver.email).then((valid) => {
+      if (!valid) {
+        form.setFieldError("caregiver.email", "Email already in use");
```
[the remark this answers is not in the document]
Comment (rethinkDeliriumRaj) [reply components/CreateAccount/Questionnaire.jsx:70]: This block is not temp - the temp part is line 7. If line 7 is changed to actual backend logic, this logic you've flagged will hold and stay.

File: components/CreateAccount/Questionnaire.jsx (hunk 3 above)
Comment (rethinkDeliriumRaj) [reply components/CreateAccount/Questionnaire.jsx:73]: you're right, good catch! setLoading is called here when it should be called inside createNewAccount's then

**Comment by rethinkDeliriumRaj** [comment]
@rethinkdeliriumjack cheated a little and did bits of both styles (bad practice) to speed up dev. 
Most of the website lives in "/" (AKA URL doesn't change when you nav between different pages) BUT the checkout_success is a different page.
More odd, the "backend" for the site's logic for producing a stripe page is included in our next.js, as opposed to with the rest of our backend on AWS.

File: pages/api/checkout_sessions/index.jsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import Stripe from "stripe";
+const stripe = new Stripe(process.env.STRIPE_SECRET_KEY, {
+  apiVersion: "2020-08-27",
+});
+
+const DOMAIN = process.env.DOMAIN ? process.env.DOMAIN : (process.env.NEXT_PUBLIC_VERCEL_URL ? `https://${process.env.NEXT_PUBLIC_VERCEL_URL}` : 'https://rethink-delirium-upload.vercel.app/');
```
Comment (rethinkDeliriumRaj) [inline pages/api/checkout_sessions/index.jsx:6]: ugly ass edit

**Comment by rethinkdeliriumjack** [comment]
image

**Comment by rethinkDeliriumRaj** [comment]
@rethinkdeliriumjack 
image

Make sure to play around w/ other dimensions before checking your PR in

**Comment by rethinkdeliriumjack** [comment]
damn wtf people have other dimensions?

**Comment by rethinkdeliriumjack** [comment]
https://github.com/rethinkDeliriumRaj/rethinkDeliriumUpload/assets/142241150/bbeaf4e9-94fc-4ef7-a8ec-ac11ebf10735

Why does it work when changes are applied, but then not work when I leave the screen

**Comment by rethinkdeliriumjack** [comment]
@rethinkDeliriumRaj Things look good to me - can I merge the PR?

**Pull request merged**